readium/navigator/src/main/java/org/readium/r2/navigator/epub/EpubNavigatorFragment.kt

```diff
@@ -306,13 +307,15 @@ class EpubNavigatorFragment internal constructor(
         _binding = ActivityR2ViewpagerBinding.inflate(inflater, container, false)
         val view = binding.root
 
+        // FIXME: If readingOrder is provided as input to the fragment, we might not want
+        // to map through the publication's readingOrder:
         positionsByReadingOrder = runBlocking { publication.positionsByReadingOrder() }
         positions = positionsByReadingOrder.flatten()
         publicationIdentifier = publication.metadata.identifier ?: publication.metadata.title
```

Comment:
We talked about this issue with @HadrienGardeur and @qnga and came up with this consensus:

Positions and total progression only make sense in the context of the publication's actual reading order. Therefore when provided with a different reading order, we should assume the positions list is empty, and also not compute the `totalProgression` when calculating the current locator.

(We might revisit this decision when introducing the guided navigations APIs, but better add more metadata than having to remove them at some point)

To avoid comparing the provided `readingOrder` to the publication one, I think you can just set the constructor parameter as a nullable `List<Link>?` and ignore the positions and total progression when it is not null.